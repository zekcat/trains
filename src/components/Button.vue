<template>
    <div>
        TrainsButton'
    </div>
</template>

<script>
export default {
    name: 'TrainsButton',

    data() {
        return {
            
        };
    },

    mounted() {
        
    },

    methods: {
        
    },
};
</script>

<style lang="scss" scoped>

</style>