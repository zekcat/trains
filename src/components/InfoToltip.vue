<template>
  <div class="info-tooltip">
    {{ title }}
  </div>
</template>

<script setup>
/**
 * @description Компонент выводит информационное сообщение для пользователя
 * при успешном выполнении действия возможной ошибке и тд
 */
import { onMounted } from "vue";

const props = defineProps({
  title: {
    type: String,
    default: "empty",
  },
  timeout: {
    type: Number,
    default: 1500,
  },
});

const emit = defineEmits(["close"]);

onMounted(() => {
  setTimeout(() => {
    emit("close");
  }, props.timeout);
});
</script>

<style lang="scss" scoped>
.info-tooltip {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-wrap: wrap;
  word-wrap: break-all;

  overflow: hidden;

  width: 40vw;
  min-height: 10vh;
  font-size: 3em;
  line-height: 100%;

  background: #f9f9fb;
  color: red;
  box-shadow: 4px 12px 51px 14px rgba(34, 60, 80, 0.64);
  border-radius: 20px;
}
</style>
