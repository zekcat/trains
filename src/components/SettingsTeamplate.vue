<template>
    <div class="settings-template">
        <button class="settings-template__btn" @click="$emit('action')">
            <slot name="button"></slot>
        </button>

        <div class="settings-template__body">
            <slot name="body"></slot>
        </div>
    </div>
</template>

<script setup>
defineEmits(['action'])
</script>

<style lang="scss" scoped>
.settings-template {
    display: flex;
    flex-direction: column;
    width: 100%;
    justify-content: center;
    align-items: flex-start;

    // .settings-template__body
    &__body {
        width: 100%;
        display: flex;
        flex-direction: column;
    }

    // .settings-template__btn
    &__btn {
        width: 100%;
        height: 35px;

        margin: 0 0 20px 0;
        font-size: 20px;
    }
}
</style>