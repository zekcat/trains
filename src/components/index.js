import SettingsTeamplate from "@/components/SettingsTeamplate.vue";

export { SettingsTeamplate };