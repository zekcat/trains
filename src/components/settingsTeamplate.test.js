import { mount } from "@vue/test-utils";
import { describe, expect, test } from "vitest";
import { SettingsTeamplate } from "@/components";

describe("unit tests", () => {
    const initComponent = () => {
        const wrapper = mount(SettingsTeamplate, {
            slots: {
                body: '<div>Test body el</div>',
                button: 'Test Button',
            }
        });

        const button = wrapper.find("button");
        return { wrapper, button };
    };

    test("check emit by click", () => {
        const { wrapper, button } = initComponent();

        button.trigger("click");
        button.trigger("click");
        const emits = wrapper.emitted();
        expect(emits).toHaveProperty("action");

        const emitUpdateByClick = wrapper.emitted("action");

        expect(emitUpdateByClick).toHaveLength(2);
        expect(emitUpdateByClick[0]).toEqual([]);
    });

    test("check slots for work", () => {
        const { wrapper } = initComponent();

        expect(wrapper.html()).toContain('<div>Test body el</div>')
        expect(wrapper.html()).toContain('Test Button')
    });

    test("snaphot teamplate on renders successfully", async () => {
        const { wrapper } = initComponent();
        expect(wrapper.html()).toMatchSnapshot();
    });
});
