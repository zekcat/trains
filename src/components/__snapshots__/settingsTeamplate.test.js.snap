// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`unit tests > snaphot teamplate on renders successfully 1`] = `
"<div data-v-f72ed292=\\"\\" class=\\"settings-template\\"><button data-v-f72ed292=\\"\\" class=\\"settings-template__btn\\">Test Button</button>
  <div data-v-f72ed292=\\"\\" class=\\"settings-template__body\\">
    <div>Test body el</div>
  </div>
</div>"
`;
